### PR TITLE
Enable fast cpp protos in the python interface.

### PR DESCRIPTION
**Pull request by keveman**

### CODE REVIEW

File: python/google/protobuf/pyext/extension_dict.h [hunk 1]
```diff
@@ -47,7 +47,8 @@ namespace protobuf {
 class Message;
 class FieldDescriptor;
 
-using internal::shared_ptr;
+using std::shared_ptr;
```
Comment (haberman) [inline python/google/protobuf/pyext/extension_dict.h]: Why did you change this? We are trying to stay compatible with C++98, which does not have shared_ptr in std.

**Comment by haberman** [comment]
Just adding a note that this is specifically for the Bazel build.

File: python/google/protobuf/pyext/extension_dict.h (hunk 1 above)
Comment (keveman) [inline python/google/protobuf/pyext/extension_dict.h]: It was complaining about internal::shared_ptr being not found. I am open to suggestions on how to fix this.

**Comment by keveman** [comment]
Hi haberman@,

I made the code compatible with C++98. It builds with both the autoconf build system and Bazel. Please take a look.

File: BUILD [hunk 2]
```diff
@@ -54,9 +54,9 @@ cc_library(
         "src/google/protobuf/stubs/time.cc",
         "src/google/protobuf/wire_format_lite.cc",
     ],
+    hdrs = glob(["src/google/protobuf/**/*.h"]),
```
Comment (haberman) [inline BUILD]: Why did you rearrange these? Is there a canonical ordering? It's not properly alphabetical before or after this change.

Comment (keveman) [inline BUILD]: That was unintentional, emacs running a tool automatically when saving.

**Comment by pherl** [comment]
LG, will defer to @haberman to take a final look.
